server/server.go

```diff
@@ -69,7 +64,7 @@ func WithReadinessChan(ch chan<- struct{}) Option {
 // NewServer creates new instance of Wasabi server
 // port - port to listen on
 // returns new instance of Server
-func NewServer(addr string, opts ...Option) *Server {
+func NewServer(addr string, serverConfig ServerConfig, opts ...Option) *Server {
```

Comment:
Let's not introduce new mandatory arguments, I think in majority of cases these timeouts don't need to be redefined and reasonable default can be use without bothering the user. I'd suggest to pass it as optional parameter.

Reply:
added an optional method